lib/widgets/molecules/recommended_mentors_filters.dart

```diff
@@ -0,0 +1,251 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:textfield_tags/textfield_tags.dart';
+import 'package:mm_flutter_app/constants/app_constants.dart';
+import 'package:mm_flutter_app/widgets/atoms/autocomplete_widget.dart';
+import 'package:mm_flutter_app/providers/explore_card_filters_provider.dart';
+
+class RecommendedMentorsFilters extends StatefulWidget {
+  final ExploreCardFiltersProvider filtersProvider;
+
+  const RecommendedMentorsFilters({super.key, required this.filtersProvider});
+
+  @override
+  State<StatefulWidget> createState() => _RecommendedMentorsFilters();
+}
+
+class _RecommendedMentorsFilters extends State<RecommendedMentorsFilters> {
+  TextfieldTagsController _countriesController = TextfieldTagsController();
+  TextfieldTagsController _languagesController = TextfieldTagsController();
+  Set<String> _selectedSkills = {};
+
+  @override
+  void initState() {
+    super.initState();
+    _countriesController = TextfieldTagsController();
+    _languagesController = TextfieldTagsController();
+    _selectedSkills = widget.filtersProvider.selectedSkills;
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+    _countriesController.dispose();
+    _languagesController.dispose();
+  }
```

Comment:
Generally, call `super.dispose()` at the very end of your implementation.

https://api.flutter.dev/flutter/widgets/State/dispose.html